java/src/main/java/eu/solven/cleanthat/engine/java/refactorer/mutators/UnnecessaryModifier.java

```diff
@@ -141,6 +141,10 @@ protected boolean processNotRecursively(NodeAndSymbolSolver<?> nodeAndSymbolSolv
 				if (isStatic) {
 					return false;
 				}
+				if (isFinal && ((MethodDeclaration) parentNode).isPrivate()
+						&& !((MethodDeclaration) parentNode).isStatic()) {
```

Comment:
Why not removing `final` if the method is static?
Could we remove final if `private || static`?

Reply:
Though static methods cannot be overridden, a static final method cannot be hidden, so for that reason alone it's not redundant. I'll link a source for that explanation.